source/views/streaming/streams/list.tsx

```diff
@@ -23,99 +24,137 @@ const styles = StyleSheet.create({
 	},
 })
 
-export const StreamListView = (): JSX.Element => {
-	let [error, setError] = React.useState<Error | null>(null)
-	let [loading, setLoading] = React.useState(true)
-	let [refreshing, setRefreshing] = React.useState(false)
-	let [streams, setStreams] = React.useState<
-		Array<{title: string; data: StreamType[]}>
-	>([])
+let groupStreams = (entries: StreamType[]) => {
+	let grouped = groupBy(entries, (j) => j.$groupBy)
+	return toPairs(grouped).map(([title, data]) => ({title, data}))
+}
 
-	React.useEffect(() => {
-		try {
-			getStreams().then(() => {
-				setLoading(false)
-			})
-		} catch (error) {
-			if (error instanceof Error) {
-				setError(error)
-			} else {
-				setError(new Error('unknown error - not an Error'))
-			}
-			return
+let filterStreams = (entries: StreamType[], filters: ListType[]) => {
+	return entries.filter((stream) => {
```

Comment:
`entries` could have a more descriptive name like `streamEntries` or `streams`. `streams.filter((stream) => { /* */ })` is a nice pattern imo.